git_test.go

```diff
@@ -159,6 +165,212 @@ func TestCloneRepo(t *testing.T) {
+
+		_ = gittest.NewRepo(t, srvFS, gittest.Commit(t, "README.md", "Hello, world!", "Wow!"))
+		key := randKeygen(t)
+		tr := gittest.NewServerSSH(t, srvFS, key.PublicKey())
+		gitURL := tr.String()
+		clientFS := memfs.New()
+
+		cloned, err := envbuilder.CloneRepo(context.Background(), envbuilder.CloneRepoOptions{
+			Path:    "/workspace",
+			RepoURL: gitURL,
+			Storage: clientFS,
+			RepoAuth: &gitssh.PublicKeys{
+				User:   "",
+				Signer: key,
+				HostKeyCallbackHelper: gitssh.HostKeyCallbackHelper{
+					// Not testing host keys here.
+					HostKeyCallback: gossh.InsecureIgnoreHostKey(),
+				},
+			},
+		})
+		// TODO: ideally, we want to test the entire cloning flow.
+		// For now, this indicates successful ssh key auth.
+		require.ErrorContains(t, err, "repository not found")
+		require.False(t, cloned)
+	})
+
+	t.Run("AuthFailure", func(t *testing.T) {
+		t.Parallel()
+
+		tmpDir := t.TempDir()
+		srvFS := osfs.New(tmpDir, osfs.WithChrootOS())
+
+		_ = gittest.NewRepo(t, srvFS, gittest.Commit(t, "README.md", "Hello, world!", "Wow!"))
+		key := randKeygen(t)
+		tr := gittest.NewServerSSH(t, srvFS, key.PublicKey())
+		gitURL := tr.String()
+		clientFS := memfs.New()
+
+		anotherKey := randKeygen(t)
+		cloned, err := envbuilder.CloneRepo(context.Background(), envbuilder.CloneRepoOptions{
+			Path:    "/workspace",
+			RepoURL: gitURL,
+			Storage: clientFS,
+			RepoAuth: &gitssh.PublicKeys{
+				User:   "",
+				Signer: anotherKey,
+				HostKeyCallbackHelper: gitssh.HostKeyCallbackHelper{
+					// Not testing host keys here.
+					HostKeyCallback: gossh.InsecureIgnoreHostKey(),
+				},
+			},
+		})
+		require.ErrorContains(t, err, "handshake failed")
+		require.False(t, cloned)
+	})
+
+	// nolint: paralleltest // t.Setenv
+	t.Run("PrivateKeyHostKeyMismatch", func(t *testing.T) {
+		t.Parallel()
+
+		tmpDir := t.TempDir()
+		srvFS := osfs.New(tmpDir, osfs.WithChrootOS())
+
+		_ = gittest.NewRepo(t, srvFS, gittest.Commit(t, "README.md", "Hello, world!", "Wow!"))
+		key := randKeygen(t)
+		tr := gittest.NewServerSSH(t, srvFS, key.PublicKey())
+		gitURL := tr.String()
+		clientFS := memfs.New()
+
+		cloned, err := envbuilder.CloneRepo(context.Background(), envbuilder.CloneRepoOptions{
+			Path:    "/workspace",
+			RepoURL: gitURL,
+			Storage: clientFS,
+			RepoAuth: &gitssh.PublicKeys{
+				User:   "",
+				Signer: key,
+				HostKeyCallbackHelper: gitssh.HostKeyCallbackHelper{
+					HostKeyCallback: gossh.FixedHostKey(randKeygen(t).PublicKey()),
+				},
+			},
+		})
+		require.ErrorContains(t, err, "ssh: host key mismatch")
+		require.False(t, cloned)
+	})
+}
+
+// nolint:paralleltest // t.Setenv for SSH_AUTH_SOCK
+func TestSetupRepoAuth(t *testing.T) {
+	t.Setenv("SSH_AUTH_SOCK", "")
+	t.Run("Empty", func(t *testing.T) {
+		opts := &envbuilder.Options{
+			Logger: testLog(t),
+		}
+		auth := envbuilder.SetupRepoAuth(opts)
+		require.Nil(t, auth)
+	})
+
+	t.Run("HTTP/NoAuth", func(t *testing.T) {
```

Comment:
Suggestion: could avoid prefix if these were nested in an otherwise empty `t.Run`. Feel free to disregard if you don't want to do the change. 😄

Reply:
I think one level of nesting is as deep as we need here in the test :-)